File: js/grade-school.js
School = function() {
  var roster = {}
  this.roster = fullRoster

  function fullRoster() {
    return roster
  };

  School.prototype.add = function (student, grade){
    if (roster[grade]==undefined){
      roster[grade]=[student]
    }
    else {
      roster[grade].push(student)
      roster[grade].sort()
    };
  };

  School.prototype.grade = function(level){
    return roster[level]==undefined ? [] : roster[level]
  };

};




module.exports = School;

File: ruby/grade_school.rb
class School
  attr_reader :roll

  def initialize
    @roll = Hash.new {|h,k| h[k]=[]}
  end

  def students(grade)
    roll[grade].sort
  end

  def add(name, grade)
    roll[grade]<<name
  end

  def students_by_grade
    full_roster = []
    roll.keys.sort.each do |grade|
      full_roster<<{grade: grade, students: roll[grade].sort}
    end
    full_roster
  end

end
